distributed/deploy/ssh.py

```diff
@@ -80,7 +79,13 @@ def __init__(
 
         self.address = address
         self.scheduler = scheduler
-        self.worker_module = worker_module
+        if worker_module != "deprecated":
+            raise ValueError(
+                "The worker_module= keyword has been deprecated. "
+                "Please use worker_class= and instead. "
+                "You will have to specify a Python class rather than a CLI command."
+            )
```

Comment:
Would it be possible to deprecate without raising an error/removing this for now?

Reply:
This would require keeping around the old implementation.  Given how rare modifying this parameter is, and the current lack of maintenance efforts, I'm inclined to make this a hard change if that is palatable to others.

Reply:
Actually, IIRC, this was specifically introduced for dask-cuda by @jacobtomlinson , which is used whenever we need an `SSHCluster` with CUDA capability: https://github.com/rapidsai/dask-cuda/blob/branch-0.15/dask_cuda/benchmarks/utils.py#L147 .

I understand your point in wanting to get rid of it and I support that, but I would prefer to have some time before that gets completely dropped. Is there an immediate use case for `worker_class`? If there isn't, I would argue in favor of keeping the current existing use case against of only supporting future uses.

Reply:
> Is there an immediate use case for worker_class?

Yes, the keyword handling is awkward otherwise.  This came up in a user issue that is referenced in the header comment of this issue.

> but I would prefer to have some time before that gets completely dropped

Yes, time makes sense.  How much time would you need?

Reply:
I'm guessing we'll have 2.19 release this week, so I'd be hoping not to drop it until 2.21-2.22, this should give us some 4 weeks to work on that. Given that we have some very pressing priorities at the moment, I don't think any of us working on dask-cuda would have time to adapt to this change before that.

Reply:
This is probably a one-line change though I would think?

```diff
-  worker_module="dask_cuda.cli.dask_cuda",
+  worker_class="dask_cuda.CUDAWorker",
```

When is the next RAPIDS release?

Reply:
Next RAPIDS release is Aug 5th but we should be able to get this done well before then.

Reply:
Sorry, I think I was misleading in https://github.com/dask/distributed/pull/3907#issuecomment-644830382 , what I meant was that I implemented it in the past and we decided not to use it to reduce the maintenance cost. So it's not just a one line change, but we will need to implement a whole new class to match this new requirement.

Reply:
Ah, I see.  We went through this process in dask-distributed where we took a lot of the logic in the CLI and moved it into the Worker/Nanny classes.  In general it was a nice win.  It made things a lot easier for us to test, and helped to reduce code duplication with other projects.

Reply:
Yes, I think we can do the same, I'll try to work on that soon, it may not be very complicated to do so, the difficult part is ensuring we don't introduce new bugs -- which we often do on such changes -- due to the spawning of new processes. This is particularly painful for UCX support, as we don't yet have full CI capabilities for multi-GPU/multi-node, so we have to test parts of it manually.

Reply:
My hope is that is mostly a copy-paste job from the existing dask-cuda CLI code and is mostly around changing how we handle inputs from Click to normal Python keywords.  *Hopefully* all of the tricky logic can be moved over without touching it too much.

Reply:
I'd be happy to take a run that this. @pentschev do you have any pointers on testing to avoid regressions.

Reply:
First and foremost we should ensure the current tests pass on a DGX-1/DGX-2, then we'll need to do some multi-node/multi-GPU testing. I'd get a PR ready first and then we sync on additional tests we should run, as we don't have many of them automated yet.